src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,688 @@
+        /// </summary>
+        /// 
+        private void GenerateNewMethodOverload(SemanticDocument semanticDocument, TExpressionSyntax expression,
+            IMethodSymbol methodSymbol, SyntaxEditor editor)
+        {
+            var document = semanticDocument.Document;
+            var syntaxFacts = semanticDocument.Document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var generator = editor.Generator;
+            var oldMethodDeclaration = expression.FirstAncestorOrSelf<SyntaxNode>(node => IsContainedInParameterizedDeclaration(node), true)!;
+            var arguments = generator.CreateArguments(methodSymbol.Parameters);
+            arguments = AddExpressionArgumentToArgumentList(arguments, expression.WithoutTrailingTrivia(), generator);
+            var memberName = methodSymbol.IsGenericMethod
+                ? generator.GenericName(methodSymbol.Name, methodSymbol.TypeArguments)
+                : generator.IdentifierName(methodSymbol.Name);
+
+            var invocation = generator.InvocationExpression(memberName, arguments);
+            List<SyntaxNode> invocationReturn;
+
+            if (methodSymbol.ReturnsVoid)
+            {
+                invocationReturn = new List<SyntaxNode>
+                {
+                    invocation
+                };
+            }
+            else
+            {
+                invocationReturn = new List<SyntaxNode>
+                {
+                    generator.ReturnStatement(invocation)
+                };
+            }
+
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol);
+            var newMethodDeclaration = generator.MethodDeclaration(newMethod, invocationReturn);
+            editor.InsertBefore(oldMethodDeclaration, newMethodDeclaration);
+        }
+
+        public static void UpdateExpressionInOriginalFunction(SemanticDocument semanticDocument,
+            TExpressionSyntax expression, SyntaxNode scope, string parameterName, SyntaxEditor editor,
+            bool allOccurrences, CancellationToken cancellationToken)
+        {
+            var generator = SyntaxGenerator.GetGenerator(semanticDocument.Document);
+            var matches = FindMatches(semanticDocument, expression, semanticDocument, scope, allOccurrences, cancellationToken);
+            var parameterNameSyntax = (TIdentifierNameSyntax)generator.IdentifierName(parameterName);
+
+            // Parenthesize the variable, and go and replace anything we find with it.
+            // NOTE: we do not want elastic trivia as we want to just replace the existing code 
+            // as is, while preserving the trivia there.
+            var replacement = generator.AddParentheses(parameterNameSyntax, includeElasticTrivia: false);
+
+            var syntaxFacts = semanticDocument.Document.GetRequiredLanguageService<ISyntaxFactsService>();
+
+            foreach (var match in matches)
+            {
+                if (match.Equals(expression))
```

Comment:
doc this.  (i get it, but it's useful for the next reader).  Effectively, you just want to say that we want to special case the replacement of hte originating expression because yadad yadda.

Reply:
added